### PR TITLE
feat(analyse): report the execution time of each spec file in order [DS-1122]

### PR DESCRIPTION
**Pull request by 0xIslamTaha**
**Description**
- Orchestrator calculates the execution time per spec. 
- Orechestror orders the results per browser based on the execution time. 

**Output Example**
```bash


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  alerts.js                                00:22       14       10        -        4        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  badges.js                                00:09        3        3        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:32       17       13        -        4        -  

Stopping design-system--master_128 ... done
Removing container_25353_chrome_0  ... done
Removing design-system--master_128 ... done
Removing network cypress_nw_brdige__master_128
generate the report ....
ExecutionTime: 
orchestratorTime: 44.247s
analyse the json report .... 
------------------------- chrome -------------------------
[
  {
    "specName": "badges.js",
    "data": [
      {
        "browser": "chrome",
        "duration": 18605
      },
      {
        "browser": "firefox",
        "duration": 0
      }
    ]
  },
  {
    "specName": "alerts.js",
    "data": [
      {
        "browser": "chrome",
        "duration": 46577
      },
      {
        "browser": "firefox",
        "duration": 0
      }
    ]
  }
]
------------------------- firefox -------------------------
[
  {
    "specName": "badges.js",
    "data": [
      {
        "browser": "chrome",
        "duration": 18605
      },
      {
        "browser": "firefox",
        "duration": 0
      }
    ]
  },
  {
    "specName": "alerts.js",
    "data": [
      {
        "browser": "chrome",
        "duration": 46577
      },
      {
        "browser": "firefox",
        "duration": 0
      }
    ]
  }
]

```

### CODE REVIEW

File: src/analyseReport.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+function orderBasedOnBrowserDuration(browser) {
+  let aDuration, bDuration;
+  return specs.sort(function (a, b) {
+    for (let i = 0; i < a.data.length; i++) {
+      if (a.data[i].browser === browser) {
+        aDuration = a.data[i].duration;
+        break;
+      }
+    }
```
Comment (RadekM86) [inline src/analyseReport.js:24]: @0xIslamTaha What is the purpose of `for` loops here? 🤔 
`aDuration` and `bDuration` will always equal the last item in the array after each loop run.
Did you want to sum the durations from specs? 🤔

File: src/analyseReport.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  specs.push({
+    specName: specName,
+    data: browsers.map((browser) => {
+      return { browser: browser, duration: 0 };
+    }),
+  });
+}
+
+function updateSpecData(suites, specName) {
+  for (let suite of suites) {
```
Comment (RadekM86) [inline src/analyseReport.js:52]: `for (let suite of suites)` could be replaced by `suites.forEach(suite => { } )`

File: src/analyseReport.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+const defaultBrowser = "chrome"; // I will use it in case of there is no browser in the title.
+
+function getBrowserFromTitle(title) {
+  title = title.toLowerCase();
+  for (let browser of browsers) {
+    if (title.includes(browser)) {
+      return browser;
+    }
+  }
+}
```
Comment (RadekM86) [inline src/analyseReport.js:14]:
```suggestion
const getBrowserFromTitle = (title) => {
  return browsers.find(browser => title.toLowerCase().includes(browser))
}
```

Comment (0xIslamTaha) [reply src/analyseReport.js:14]: done

File: src/analyseReport.js (hunk 1 above)
Comment (0xIslamTaha) [reply src/analyseReport.js:24]: @RadekM86 
I need to return the duration based on the browser value. `a.data = [ {browser: 'chrome', duration: 1}, {browser: 'firefox', duration: 2} ]`. I updated the implementation using the filter method.

File: src/analyseReport.js (hunk 2 above)
Comment (0xIslamTaha) [reply src/analyseReport.js:52]: done

File: src/analyseReport.js [hunk 4]
```diff
@@ -1,71 +1,42 @@
...
     return aDuration - bDuration;
   });
 }
 
 function intiateSpecData(specName) {
-  for (let spec of specs) {
-    if (spec["specName"] === specName) {
-      return;
-    }
+  if (!specs.find( spec => spec.specName === specName )) {
```
Comment (RadekM86) [inline src/analyseReport.js:19]: @0xIslamTaha for creating arrays or adding to array you can spread `...` array and use lodash uniq to make it unique. Or you can try Array.map or Array.forEach 🤔

File: src/analyseReport.js [hunk 5]
```diff
@@ -1,71 +1,42 @@
...
-    }
-
-    for (let i = 0; i < b.data.length; i++) {
-      if (b.data[i].browser === browser) {
-        bDuration = b.data[i].duration;
-        break;
-      }
-    }
-
+    let aDuration = a.data.find( item => item.browser === browser ).duration
```
Comment (RadekM86) [inline src/analyseReport.js:12]: This looks way better than `for` loop 💪

**Review by RadekM86 — APPROVED** [review]
Overall it looks ok 👍 
I didn't have the opportunity to test run this code though

File: src/analyseReport.js (hunk 4 above)
Comment (0xIslamTaha) [reply src/analyseReport.js:19]: @RadekM86
I need to intialize it only for the first time so that is why I'm using find and map

**Review by 0xIslamTaha — COMMENTED** [review]
> Overall it looks ok 
> I didn't have the opportunity to test run this code though

I tested it and we r going to use it now :D, let's see

File: src/analyseReport.js (hunk 4 above)
Comment (RadekM86) [reply src/analyseReport.js:19]: Ok 👍

**Pull request merged**